src/integration/java/io/pinecone/integration/dataPlane/QueryErrorTest.java

```diff
@@ -0,0 +1,164 @@
+package io.pinecone.integration.dataPlane;
+
+import com.google.common.util.concurrent.ListenableFuture;
+import io.grpc.StatusRuntimeException;
+import io.pinecone.clients.AsyncIndex;
+import io.pinecone.clients.Index;
+import io.pinecone.configs.PineconeConnection;
+import io.pinecone.exceptions.PineconeValidationException;
+import io.pinecone.helpers.RandomStringBuilder;
+import io.pinecone.proto.DescribeIndexStatsResponse;
+import io.pinecone.proto.VectorServiceGrpc;
+import io.pinecone.unsigned_indices_model.QueryResponseWithUnsignedIndices;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static io.pinecone.helpers.BuildUpsertRequest.generateSparseIndicesByDimension;
+import static io.pinecone.helpers.BuildUpsertRequest.generateVectorValuesByDimension;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class QueryErrorTest {
+
+    private static Index index;
+    private static AsyncIndex asyncIndex;
+    private static final int dimension = 3;
+
+    @BeforeAll
+    public static void setUp() throws IOException, InterruptedException {
+        PineconeConnection connectionMock = mock(PineconeConnection.class);
+
+        VectorServiceGrpc.VectorServiceBlockingStub stubMock = mock(VectorServiceGrpc.VectorServiceBlockingStub.class);
+        VectorServiceGrpc.VectorServiceFutureStub asyncStubMock = mock(VectorServiceGrpc.VectorServiceFutureStub.class);
+
+        when(connectionMock.getBlockingStub()).thenReturn(stubMock);
+        when(connectionMock.getFutureStub()).thenReturn(asyncStubMock);
+
+        index = new Index(connectionMock);
+        asyncIndex = new AsyncIndex(connectionMock);
```

Comment:
I'm confused why we're mocking here. Should we be testing the actual calls for integration testing purposes?

You should still be able to call without mocking and assert on the expected errors, and then we wouldn't need to disable the tests around server-side validations.

Reply:
For error tests, we want to verify client-side validations only, so there's no need to hit the server. One option for server side error tests I thought was to move it to the query integration test but since that test is meant for successful querying cases only, it wouldn't be ideal to place it there. Also the disabled test is verifying server side error logic so its not something client tests should ideally worry about.

Reply:
The client-side validations will prevent the requests from being sent to the server anyways. If the disabled tests aren't even meant to be run then I wouldn't include them in here if you're keeping the mocking paradigm, it's sort of confusing otherwise.

We assert and validate on server responses in integration tests on the other client, I don't think there's anything wrong in doing that. I did a bunch of that in create index error tests in the control plane: https://github.com/pinecone-io/pinecone-java-client/blob/7a26bde5d5c67ec64476b1e3b4a2fa6f0049febc/src/integration/java/io/pinecone/integration/controlPlane/pod/CreateDescribeListAndDeleteIndexTest.java#L135

Reply:
> The client-side validations will prevent the requests from being sent to the server anyways.
I'm actually running a test that queries with a vector of dimension 1 for an index with dimension 3, so that's not something we check on client side. That's why the above statement is not true. We should hit the server if its not needed. And in my case, I'll move the server-side validations error test to the query integration test, so I can still mock stubs and run the client side validation tests.

Reply:
Turns out this is not even an issue for serverless, so we are allowed to query an index of dimension d1 with input vector of dimension d2. Will remove these tests and follow up with the team.